client/src/util/fetchFromAPI.js

```diff
@@ -1,15 +1,19 @@
 export const queryAPI = async (options) => {
-  const { url, onSuccess, onError, method } = options;
+  const { url, onSuccess, onError, method, toast } = options;
   const result = {
     data: null,
     error: null,
   };
-  const response = fetch(url, method);
+  const response = await fetch(url, method);
   if (response.ok) {
     result.data = await response.json();
+    //trigger success toast on success
+    toast.success();
```

Comment:
Thinking of this from a design perspective, do you always want this fetch utility to open a toast on success, or does that belong at the call site instead?

Same question for the error path I suppose.